src/lib/components/Header.svelte

```diff
@@ -1,54 +1,178 @@
+
+  const animate = (node, args) =>
+    args.enable
+      ? fly(node, { y: !scrollingDown ? -args.y : args.y, duration: 300 })
+      : undefined;
+
+  if (browser) {
+    updateScrollPos();
+  }
+
+  onMount(() => {
+    if (browser) {
+      window.addEventListener('scroll', updateScrollPos);
+    }
+  });
+
+  function updateScrollPos() {
+    scrollingDown = window.scrollY > scrollPos;
+    scrolledDown = window.scrollY !== 0;
+    scrollPos = window.scrollY;
+  }
 </script>
 
-<header>
-  <nav>
-    <div class="home">
-      <a sveltekit:prefetch href="/" class:active={onExplore}>Explore</a>
-      <a
-        href="/dashboard"
-        on:click={() => goto(`/dashboard`)}
-        class:active={onDashboard}>Dashboard</a
+<header
+  class:hide
+  style:box-shadow={scrolledDown && !hide ? 'var(--shadow)' : ''}
+>
+  <div class="inner">
+    {#if showCustomHeaderContent && $headerContent.component}
+      <div
+        transition:animate={{ enable: headerContent, y: 20 }}
+        class="content page"
+      >
+        <svelte:component
+          this={$headerContent.component}
+          {...$headerContent.props}
+        />
+      </div>
+    {:else}
+      <div
+        transition:animate={{ enable: !!$headerContent.component, y: 20 }}
+        class="content default"
       >
-    </div>
-    <div class="user">
-      <Connect />
-    </div>
-  </nav>
+        <nav>
+          <a sveltekit:prefetch href="/" class:active={onExplore}>Explore</a>
+          <a
+            sveltekit:prefetch
+            href="/dashboard"
+            on:click={() => goto(`/dashboard`)}
+            class:active={onDashboard}>Dashboard</a
+          >
+        </nav>
+        <div class="buttons">
+          {#if $connectedAndLoggedIn && onDashboard}
+            <div
+              in:fly={{ y: 10, duration: 300, delay: 300 }}
+              out:fly={{ y: 10, duration: 300 }}
+              class="create-button"
+            >
+              <Button
+                icon={TokenStreams}
+                on:click={() => modal.show(CreateModal)}
+                >Create workstream</Button
+              >
+            </div>
+          {/if}
+          <div class="user">
+            <Connect />
+          </div>
+        </div>
+      </div>
+    {/if}
+  </div>
 </header>
 
+<div class="spacer" />
+
 <style>
-  nav {
+  header {
+    top: 0;
+    left: 0;
+    right: 0;
+    height: 72px;
+    box-sizing: border-box;
+    background-color: var(--color-background);
+    z-index: 10;
+    position: fixed;
+    transition: box-shadow 0.3s, transform 0.3s;
+    --shadow: 0px 6px 16px rgba(0, 0, 0, 0.05),
+      0px 2.77398px 7.39728px rgba(0, 0, 0, 0.0370838),
+      0px 1.58721px 4.23256px rgba(0, 0, 0, 0.031339),
+      0px 0.963424px 2.56913px rgba(0, 0, 0, 0.0269974),
+      0px 0.580506px 1.54802px rgba(0, 0, 0, 0.0230026),
+      0px 0.323263px 0.862035px rgba(0, 0, 0, 0.018661),
+      0px 0.139033px 0.370755px rgba(0, 0, 0, 0.0129162);
```

Comment:
What's this hard coded shadow? Wouldn't we use the design system one?

Reply:
Just landed in `main`: https://github.com/radicle-dev/radicle-upstream/commit/dd7f1b165befe01c42b5ed1253696b7bd24688a4.

Reply:
For some reason it wasn't being imported :/ Let me look into that